AsyncEnumerable.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Runtime.CompilerServices;
+
+namespace MyApp
+{
+    internal static partial class AsyncEnumerable
+    {
+        public static IAsyncEnumerable<T> Empty<T>() => EmptyAsyncEnumerable<T>.Instance;
+
+        class EmptyAsyncEnumerable<T> : IAsyncEnumerable<T>
+        {
+            public static readonly IAsyncEnumerable<T> Instance = new EmptyAsyncEnumerable<T>();
+
+            private EmptyAsyncEnumerable() { }
+
+            public IAsyncEnumerator<T> GetAsyncEnumerator(CancellationToken cancellationToken = default) => EmptyAsyncEnumerator<T>.Instance;
+        }
+
+        class EmptyAsyncEnumerator<T> : IAsyncEnumerator<T>
+        {
+            public static readonly IAsyncEnumerator<T> Instance = new EmptyAsyncEnumerator<T>();
+
+            private EmptyAsyncEnumerator() { }
+
+            public T Current => default;
+
+            public ValueTask<bool> MoveNextAsync() => new ValueTask<bool>(false);
+
+            public ValueTask DisposeAsync() => default;
+        }
+
+        public static async Task<bool> Any<T>(this IAsyncEnumerable<T> source)
```

Comment:
Looks like we just duplicated this from https://github.com/dotnet/dotnet/blob/2db1f5ee2bdda2e8d873769325fabede32e420e0/src/runtime/src/libraries/System.Linq.AsyncEnumerable/src/System/Linq/AnyAsync.cs#L23C13-L25C52 

because we don't have access to the version where they added this to linq.

Could we just copy directly from their impl? (so we get cancellation as well)